mozilla-services/socorro
# bug 1657668, 1558916: rework product support

This reworks product support in Crash Stats. Previously, it was split across a DB table and a bunch of hard-coded stuff across the codebase. Now it's based solely on product settings files in `product_details/` directory.

This makes it easier for other people to submit changes to product settings--just edit the file in GitHub. Further, this reduces maintenance--we don't have to maintain database entries across multiple environments.

One big change here is that `featured_versions` changes now ride the deploys--Crash Stats no longer looks at GitHub for data.

This PR provides:

1. infrastructure for the new `product_details/` -based support
2. a test to make sure changes to product settings files are correct so we can catch issues in CI
3. rework product home page to use the new system and drop the `crashstats_product` table
4. rework featured versions figuring to use the new system (and fix bug 1558916)

To test:

1. (wait for cache to expire)
2. visit home page
3. visit product home pages--make sure featured versions are correct for each one
4. visit supersearch page
